src/aead/chacha20_poly1305.rs

```diff
@@ -278,6 +279,67 @@ mod tests {
                     (CRYPTO_POLY1305_STATE_LEN + 255) / 256);
     }
 
+    #[test]
+    pub fn chacha20_tests(){
```

Comment:
I feel like this test isn't documented clearly enough. If I didn't already know what it was supposed to do, I would have trouble figuring it out, I think.

The primary point of this test, especially the end part, is to test that the `ChaCha20_ctr32` function properly handles input and output buffers that overlap to various degrees, for various alignment factors. This is an important test because on some platforms the `ChaCha20_ctr32` code doesn't work for partially-overlapping buffers and we have to use a `memmove` to move the input to the output buffer so that they overlap exactly. OpenSSL and BoringSSL don't even try to use `ChaCha20_ctr32` this way so we're on our own when it comes to testing this.

Let's find a way to capture all of the above in a comment.